### PR TITLE
[Hacktoberfest] Migrate docs to jenkinsci repo

### PR DESCRIPTION
**Pull request by Dan-Heath**
This PR migrates the MQ Notifier plugin documentation from the **plugins-wiki-docs** repo to the **mq-notifier-plugin** repo as part of the Hacktoberfest docs-to-code project. It includes the following changes:

- Merges the content from the wiki into the README.md file
- Creates a CHANGELOG.md file and copies the version history information from the wiki into it
- Updates the pom.xml file with the new URL for the docs
- Makes some minor formatting edits for consistency and flow

Please add the **hacktoberfest-accepted** label if you approve this PR. Thank you!

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did

### CODE REVIEW

**Comment by Dan-Heath** [comment]
Hello @TWestling , I created this PR as part of Hacktoberfest. I think it is ready for review. Please take a look and let me know if you have any questions or if you'd like me to make any changes. Thank you!

File: pom.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
   <name>MQ Notifier</name>
   <description>Publishes messages on an MQ server</description>
-  <url>https://wiki.jenkins-ci.org/display/JENKINS/MQ+Notifier+Plugin</url>
+  <url>https://github.com/jenkisnci/mq-notifier-plugin</url>
```
Comment (magnusbaeck) [inline pom.xml]: Typo: jenkisnci ⟶ jenkinsci

Comment (Dan-Heath) [reply pom.xml]: > Typo: jenkisnci ⟶ jenkinsci

Thank you @magnusbaeck ! I fixed it.

**Comment by TWestling** [comment]
@Dan-Heath Sorry for the delay in looking at this, and thanks for the PR!

**Pull request merged**